posthog-node/src/posthog-node.ts

```diff
@@ -107,12 +107,23 @@ export class PostHog implements PostHogNodeV1 {
     return this._sharedClient.optOut()
   }
 
-  capture({ distinctId, event, properties, groups, sendFeatureFlags }: EventMessageV1): void {
+  capture({ distinctId, event, properties, groups, sendFeatureFlags, timestamp }: EventMessageV1): void {
     this.reInit(distinctId)
     if (groups) {
       this._sharedClient.groups(groups)
     }
-    this._sharedClient.capture(event, properties, sendFeatureFlags || false)
+
+    const _capture = (): void => {
```

Comment:
moving this here works.

Talk to me a bit about the race condition issues?

Is it related to other calls to `reInit` changing the flags because the distinctID changes? Since the `capture` call is not blocking?

Hmm, that makes sense

Reply:
Yeah exactly. It's one of the things I never really nailed with this combined JS approach 😢 

I think the code will work still if we just change it to use the returned value of the async call?

Reply:
Hmm, okay other flag calls are blocking, so this isn't an issue.